modules/api/VcenterConnection.py

```diff
@@ -0,0 +1,152 @@
+from interfaces.host import Host
+from interfaces.vcenter import Vcenter
+
+from pyVmomi import vim, vmodl
+from pyVim.connect import SmartConnect, SmartConnectNoSSL, Disconnect
+
+from typing import List
+
+import logging
+import socket
+
+# Init Logging
+logger = logging.getLogger('esxi-exporter')
+
+
+class VcenterConnection:
+    def __init__(self, host: str, user: str, password: str, verify_ssl: bool = False):
+        """
+        Provides functionality to fetch overall state of esxi host-systems.
+
+        :param host: The url to the vcenter without https. You must use url not
+        ip-address.
+        :param user: The username used to login
+        :param password: The password used to login
+        :param verify_ssl: Should ssl certificates be verified? Default = False.
+        """
+
+        self.api = None
+        self.host = host
+        self.user = user
+        self.password = password
+        self.verify_ssl = verify_ssl
+        if verify_ssl:
+            self._connect_class = SmartConnect
+        else:
+            logger.warning('vCenter connection with ssl disabled: %s' % host)
+            self._connect_class = SmartConnectNoSSL
+
+    def login(self) -> None:
+        logger.debug('vCenter logging in: %s' % self.host)
+        try:
+            self.api = self._connect_class(protocol='https', host=self.host,
+                                           user=self.user, pwd=self.password)
+            logger.debug('successfully logged into vCenter: %s' % self.host)
+
+        except socket.gaierror:
+            message = 'Vcenter: DNS error, could not resolve name: %s' % self.host
+            logger.error(message)
+            raise

```

Comment:
These `raises` will terminate your entire run, shouldn't we just skip this one VC? Maybe returning False would do the trick and evaluating if the login succeeded at caller side.

Reply:
login returns true or false now.

Reply:
No raise anymore